src/activities/Elsa.Activities.Rpa.Web/Extensions/SeleniumExtensions.cs

```diff
@@ -0,0 +1,355 @@
+
+                if (html == null) continue;
+                doc.LoadHtml(html);
+                nodes = doc.DocumentNode.Descendants().AsQueryable();
+            }
+            return nodes;
+        }
+        public static HtmlNode GetHtmlNode(this ISearchContext element, Func<HtmlNode, bool> query, int index = 0, TimeSpan timeout = default, ILogger logger = default)
+        {
+            if (timeout == default)
+                timeout = TimeSpan.FromSeconds(20);
+            if (index < 0)
+                throw new ArgumentOutOfRangeException("index must be positive or zero");
+            var doc = new HtmlDocument();
+            var start = DateTime.Now;
+
+            HtmlNode node = null;
+            while (node == null && (DateTime.Now - start) < timeout)
+            {
+                try
+                {
+                    logger?.LogTrace($"before getting page source");
+                    var html = "";
+                    if (element as IWebElement != null)
+                    {
+                        var outerHtml = ((IWebElement)element).GetAttribute("outerHtml");
+                        if (outerHtml == null)
+                            outerHtml = ((IWebElement)element).GetAttribute("outerHTML");
+                        if (outerHtml == null) continue;
+                        html = outerHtml;
+                    }
+                    if (element as IWebDriver != null)
+                        html = ((IWebDriver)element).PageSource;
+
+                    if (html == null) continue;
+                    else
+                        logger?.LogTrace($"page source loaded: {html}");
+                    doc.LoadHtml(html);
+                    node = doc.DocumentNode.Descendants()
+                        .Where(query)
+                        .Skip(index - 1)
+                        .FirstOrDefault();
+                }
+                finally
+                {
+                    Task.Delay(1000).GetAwaiter().GetResult();
+                }
+            }
+            if (node == null)
+                throw new Exception("element not found. check searching criterias");
+            return node;
+        }
+        #region FindElement
+        public static IWebElement FindElement<TChild>(this IWebElement element, Func<HtmlNode, bool> query, int index = 0, TimeSpan timeout = default, ILogger logger = default)
+        {
+            var node = GetHtmlNode(element, query, index, timeout, logger);
+            if (node == null)
+                throw new Exception("element not found. check searching criterias");
+            return element.FindElement(By.XPath(GetXPath(node)));
+        }
+        /// <summary>
+        /// per usarlo è necessario importare il pacchetto HtmlAgilityPack
+        /// </summary>
+        /// <param name="element"></param>
+        /// <param name="query"></param>
+        /// <param name="index"></param>
+        /// <returns></returns>
```

Comment:
Please translate to English (en-US) and remove the empty XML elements. They just add noise.